contracts/helpers/MultiBlockHarvester.sol

```diff
@@ -0,0 +1,202 @@
+                                                        TRUSTED FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Initiate a rebalance
+     * @param scale scale to apply to the rebalance amount
+     * @param yieldBearingAsset address of the yieldBearingAsset
+     */
+    function harvest(address yieldBearingAsset, uint256 scale, bytes calldata) external onlyTrusted {
+        if (scale > 1e9) revert InvalidParam();
+        YieldBearingParams memory yieldBearingInfo = yieldBearingData[yieldBearingAsset];
+        (uint8 increase, uint256 amount) = _computeRebalanceAmount(yieldBearingAsset, yieldBearingInfo);
+        amount = (amount * scale) / 1e9;
+        if (amount == 0) revert ZeroAmount();
+
+        try transmuter.updateOracle(yieldBearingAsset) {} catch {}
+        _rebalance(increase, yieldBearingAsset, yieldBearingInfo, amount);
+    }
+
+    /**
+     * @notice Finalize a rebalance
+     * @param yieldBearingAsset address of the yieldBearingAsset
+     */
+    function finalizeRebalance(address yieldBearingAsset, uint256 balance) external onlyTrusted {
+        try transmuter.updateOracle(yieldBearingAsset) {} catch {}
+        _adjustAllowance(yieldBearingAsset, address(transmuter), balance);
+        uint256 amountOut = transmuter.swapExactInput(
+            balance,
+            0,
+            yieldBearingAsset,
+            address(agToken),
+            address(this),
+            block.timestamp
+        );
+        address depositAddress = yieldBearingAsset == XEVT
+            ? yieldBearingToDepositAddress[yieldBearingAsset]
+            : address(0);
+        _checkSlippage(balance, amountOut, yieldBearingAsset, depositAddress, true);
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        INTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    function _rebalance(
+        uint8 typeAction,
+        address yieldBearingAsset,
+        YieldBearingParams memory yieldBearingInfo,
+        uint256 amount
+    ) internal {
+        if (amount > maxOrderAmount) revert TooBigAmountIn();
+        _adjustAllowance(address(agToken), address(transmuter), amount);
+        address depositAddress = yieldBearingToDepositAddress[yieldBearingAsset];
+        if (typeAction == 1) {
+            uint256 amountOut = transmuter.swapExactInput(
+                amount,
+                0,
+                address(agToken),
+                yieldBearingInfo.depositAsset,
+                address(this),
+                block.timestamp
+            );
+            _checkSlippage(amount, amountOut, yieldBearingInfo.depositAsset, depositAddress, false);
```

Comment:
clearly confusing names
`yieldBearingInfo.depositAsset` is the token rebalancing from/to yiled bearing token
`depositAddress` is where to deposit funds to get the yield bearing

Let's rename such that this is clear